src/mococrw/padding_mode.h

```diff
@@ -0,0 +1,214 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include "key.h"
+
+namespace mococrw {
+/**
+ * @brief RSAPadding
+ *
+ * This class defines the interface for RSA Padding Mode objects, which containin all the parameters
+ * specific to a given padding mode used by encryption/decription and sign/verify operations.
+ */
+class RSAPadding
+{
+public:
+    /**
+     * @brief Get the padding mode
+     *
+     * Virtual getter method for the padding mode.
+     */
+    virtual openssl::RSAPaddingMode getPadding() const = 0;
+};
+
+/**
+ * @brief NoPadding
+ *
+ * This class defines the parameters specific to the RSA no padding mode.
+ */
+class NoPadding: public RSAPadding {
+public:
+    openssl::RSAPaddingMode getPadding() const {
```

Comment:
Please use the virtual keyword and override, if you are instantiating the virtual function "virtual openssl::RSAPaddingMode getPadding() const = 0;"  from class RSAPadding.

Reply:
Done